src/order_backup.cpp

```diff
@@ -54,14 +52,8 @@ OrderBackup::OrderBackup(const Vehicle *v, uint32_t user) : user(user), tile(v->
 		this->clone = (v->FirstShared() == v) ? v->NextShared() : v->FirstShared();
 	} else {
 		/* Else copy the orders */
-		Order **tail = &this->orders;
-
-		/* Count the number of orders */
-		for (const Order *order : v->Orders()) {
-			Order *copy = new Order();
-			copy->AssignOrder(*order);
-			*tail = copy;
-			tail = &copy->next;
+		for (const Order &order : v->Orders()) {
+			this->orders.emplace_back(order);
 		}
```

Comment:
Won't something simpler, like this, work as well?
```suggestion
		this->orders = v->Orders();
```